models/engine.go

```diff
@@ -38,8 +64,30 @@ func (r *RestartableEngine) Ready() <-chan struct{} {
 }
 
 func (r *RestartableEngine) Run(ctx context.Context) error {
-	// todo add restart logic
-	return r.engine.Run(ctx)
+	var err error
+	for i, b := range r.backoff {
+		<-time.After(b) // wait for it
```

Comment:
maybe add a select to exit early if the context is cancelled

Reply:
oh nice yeah